### PR TITLE
Add missing error check

### PR DESCRIPTION
**Pull request by mrkm4ntr**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Add missing error check.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/controller/podautoscaler/horizontal.go [hunk 1]
```diff
@@ -342,6 +342,9 @@ func (a *HorizontalController) reconcileKey(key string) (deleted bool, err error
 		delete(a.recommendations, key)
 		return true, nil
 	}
+	if err != nil {
```
Comment (draveness) [inline pkg/controller/podautoscaler/horizontal.go]: Looks Good

**Review by draveness — COMMENTED** [review]
/ok-to-test

**Comment by mrkm4ntr** [comment]
@MaciekPytel Could you review this?

**Review by draveness — APPROVED** [review]
/lgtm
/remove-lifecycle stale

**Review by janetkuo — APPROVED** [review]

**Pull request merged**